packages/settings-library/src/settings_library/base.py

```diff
@@ -22,41 +15,33 @@
 ] = "%s auto_default_from_env unresolved, defaulting to None"
 
 
-class DefaultFromEnvFactoryError(ValidationError):
-    ...
+class DefaultFromEnvFactoryError(ValueError):
+    def __init__(self, errors):
+        super().__init__()
+        self.errors = errors
 
 
-def create_settings_from_env(field: ModelField):
+def _create_settings_from_env(field_name: str, info: FieldInfo):
     # NOTE: Cannot pass only field.type_ because @prepare_field (when this function is called)
     #  this value is still not resolved (field.type_ at that moment has a weak_ref).
     #  Therefore we keep the entire 'field' but MUST be treated here as read-only
 
     def _default_factory():
         """Creates default from sub-settings or None (if nullable)"""
-        field_settings_cls = field.type_
+        field_settings_cls = get_type(info)
         try:
             return field_settings_cls()
 
         except ValidationError as err:
-            if field.allow_none:
+            if is_nullable(info):
                 # e.g. Optional[PostgresSettings] would warn if defaults to None
                 _logger.warning(
                     _DEFAULTS_TO_NONE_MSG,
-                    field.name,
+                    field_name,
                 )
                 return None
-
-            def _prepend_field_name(ee: ErrorList):
-                if isinstance(ee, ErrorWrapper):
-                    return ErrorWrapper(ee.exc, (field.name, *ee.loc_tuple()))
-                assert isinstance(ee, Sequence)  # nosec
-                return [_prepend_field_name(e) for e in ee]
-
-            raise DefaultFromEnvFactoryError(
-                errors=_prepend_field_name(err.raw_errors),
-                model=err.model,
-                # FIXME: model = shall be the parent settings?? but I dont find how retrieve it from the field
-            ) from err
+            _logger.warning("Validation errors=%s", err.errors())
+            raise DefaultFromEnvFactoryError(errors=err.errors()) from err
```

Comment:
Have you checked if the new errors is stil readable like the one before? Just make sure that is still the case.